signer/src/wsts_state_machine.rs

```diff
@@ -30,33 +39,435 @@ use wsts::traits::Signer as _;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl std::ops::DerefMut for FireCoordinator {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+/// Wrapper for a WSTS FROST coordinator state machine.
+#[derive(Debug, Clone, PartialEq)]
+pub struct FrostCoordinator(frost::Coordinator<Aggregator>);
+
+impl std::ops::Deref for FrostCoordinator {
+    type Target = frost::Coordinator<Aggregator>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl std::ops::DerefMut for FrostCoordinator {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+/// A trait for WSTS state machines.
+pub trait WstsCoordinator
+where
+    Self: Sized,
+{
+    /// Creates a new coordinator state machine.
+    fn new<I>(signers: I, threshold: u16, message_private_key: PrivateKey) -> Self
+    where
+        I: IntoIterator<Item = PublicKey>;
+
+    /// Gets the coordinator configuration.
+    fn get_config(&self) -> Config;
+
+    /// Save the state required to reconstruct the state machine.
+    fn save(&self) -> SavedState;
+
+    /// Create a new coordinator state machine from the given aggregate
+    /// key.
+    ///
+    /// # Notes
+    ///
+    /// The `WstsCoordinator` is a state machine that is responsible for
+    /// DKG and for facilitating signing rounds. When created the
+    /// `WstsCoordinator` state machine starts off in the `IDLE` state,
+    /// where you can either start a signing round or start DKG. This
+    /// function is for loading the state with the assumption that DKG has
+    /// already been successfully completed.
+    fn load<S>(
+        storage: &S,
+        aggregate_key: PublicKeyXOnly,
+        signer_public_keys: impl IntoIterator<Item = PublicKey> + Send,
+        threshold: u16,
+        signer_private_key: PrivateKey,
+    ) -> impl Future<Output = Result<Self, error::Error>> + Send
+    where
+        S: storage::DbRead + Send + Sync;
+
+    /// Process the given message.
+    fn process_message(
+        &mut self,
+        message: &Message,
+    ) -> Result<(Option<Packet>, Option<OperationResult>), Error> {
+        let packet = Packet::from_message(message);
+        self.process_packet(&packet)
+    }
+
+    /// Process inbound messages
+    fn process_inbound_messages(
+        &mut self,
+        messages: &[Message],
+    ) -> Result<(Vec<Packet>, Vec<OperationResult>), Error> {
+        let packets = messages
+            .iter()
+            .map(Packet::from_message)
+            .collect::<Vec<_>>();
+        self.process_inbound_packets(&packets)
+    }
+
+    /// Process the given packet.
+    fn process_packet(
+        &mut self,
+        packet: &Packet,
+    ) -> Result<(Option<Packet>, Option<OperationResult>), Error>;
+
+    /// Process inbound packets
+    fn process_inbound_packets(
+        &mut self,
+        packets: &[Packet],
+    ) -> Result<(Vec<Packet>, Vec<OperationResult>), Error>;
```

Comment:
This is only used in `process_inbound_messages`, so if that is not used this is also not used

Reply:
Same as above comment, removed in [cb95417](https://github.com/stacks-network/sbtc/pull/1287/commits/cb9541747dc90d6fc486f3308536ebddc830b112)